### PR TITLE
#2145-settings-page-tabs

### LINKED ISSUE #2145: [Frontend] - Switch Settings page into the new tab based design
### Description

Currently, the settings page is a single tab with page blocks. Convert the existing settings page into a tab based view. The page should include a tab pane with Details as the default tab, with all the other components in the subsequent tabs. Do not remove the page blocks on the components. Just move the entire page block to the new tab.

### Acceptance Criteria

- [ ] The settings page should be divided into the details and preferences pane.
- [ ] Clicking on a new tab on the tabs pane should switch to the specific tool.
- [ ] The functionality of the admin tools should remain the same, especially the visibility of tools based on user role.

### Proposed Solution

- Look at how this is being done in Change Requests and Projects.
- Look at the mocks below on which component should be on which tab.
- Keep in mind there are some features on the mocks that haven't been built yet like the schedule settings.
- Since you are just moving the components without removing the page blocks it is okay if it doesn't look like the mocks, just make sure the right information is on the right tab.

### Mocks

image
image

### PR DESCRIPTION
**Pull request by alexpark0**
## Changes

Changed settings page into a tab-based view

## Screenshots
image

image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2145

### CODE REVIEW

**Review by ryanhowe28 — CHANGES_REQUESTED** [review]
Looking good, small things

File: src/frontend/src/pages/SettingsPage/Preferences.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    setShowAlert(true);
+    setTimeout(() => {
+      auth.signout();
+    }, 2000);
+  };
+
+  return (
+    <PageLayout title="Preferences">
+      {showAlert && <Alert severity="info">Haha {auth.user?.firstName} bye bye!</Alert>}
+      <PageBlock title={'Organization Settings'}>
```
Comment (ryanhowe28) [inline src/frontend/src/pages/SettingsPage/Preferences.tsx:106]: Lets move organization settings to the details tab

File: src/frontend/src/pages/SettingsPage/SettingsPreferences.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    secureSettingsIsLoading ||
+    !userSecureSettings ||
+    allTeamsIsLoading ||
+    !teams
+  )
+    return <LoadingIndicator />;
+
+  return (
+    <PageLayout title="Preferences">
+      {showAlert && <Alert severity="info">Haha {auth.user?.firstName} bye bye!</Alert>}
```
Comment (ryanhowe28) [inline src/frontend/src/pages/SettingsPage/SettingsPreferences.tsx:46]: Looks like you flipped the names/content of the preferences and details files. In the middle of that you took this line and put it on the preference page when it should be on the details page and should show when someone hits trickster mode. Look at how it was before

**Review by ryanhowe28 — CHANGES_REQUESTED** [review]
One thing that got put out of place.

**Review by ryanhowe28 — DISMISSED** [review]
LGTM

File: src/frontend/src/pages/SettingsPage/SettingsDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          <DetailDisplay label="Role" content={displayEnum(user.role)} />
+        </Grid>
+        <Grid item xs={12} sm={6} md={4} lg={2}>
+          <DetailDisplay
+            label="Teams"
+            content={userTeams.length === 0 ? 'None' : userTeams.map((team) => team.teamName).join(', ')}
+          />
+        </Grid>
+      </Grid>
+      <PageBlock title={'Organization Settings'}>
```
Comment (walker-sean) [inline src/frontend/src/pages/SettingsPage/SettingsDetails.tsx:130]: I know this has gone back and forth, but lets put everything that can be changed into the preferences tab. The details tab should just be that readonly list

**Review by walker-sean — CHANGES_REQUESTED** [review]
Looking good!

We also need to add the schedule settings (in develop), and there are some merge conflicts

**Pull request merged**